brave/brave-core
# WebTorrent Support

Fix brave/brave-browser#289.

What's included in this PR:
- WebTorrent support of magnet links

What's not included in this PR:
- WebTorrent support of .torrent links (https://github.com/brave/brave-browser/issues/855)
- Support of trackless torrents using bittorrent-dht/client (https://github.com/brave/brave-browser/issues/857)
- l10n (https://github.com/brave/brave-browser/issues/858)

## Submitter Checklist:


## Test Plan:
Manual Test:
1. Open a new tab and visit https://webtorrent.io/free-torrents, click magnet link of Big Buck Bunny should leads to a torrent viewer page without start downloading
2. Click start download, torrent should start to be downloading
3. Click any link under save file should be able to download that specific file
4. Click on poster.jpg, should open a tab and show the image directly
5. Go back to the torrent downloading page and click on Big Buck Bunny.mp4, should open a tab to stream the media
(Note: It might take a while for the media to be ready to play.)
6. Close the media streaming tab and go back to the torrent downloading page
7. Go back to https://webtorrent.io/free-torrents tab and see if it also works by right click the magnet link to open it in a new tab

Automation Test:
`npm run test -- brave_browser_tests --filter=BraveContentBrowserClientTest.RewriteMagnetURL*`
`npm run test-unit`

## Reviewer Checklist:
